src/Microsoft.TestPlatform.ObjectModel/Utilities/AssemblyHelper.cs

```diff
@@ -315,32 +315,21 @@ internal static Version GetTargetFrameworkVersionFromVersionString(string versio
         /// <summary>
         /// When test run is targeted for .Net4.0, set target framework of test appdomain to be v4.0.
         /// With this done tests would be executed in 4.0 compatiblity mode even when  .Net4.5 is installed.
-        /// This needs to be done using reflection, as this dll is compiled for .Net3.5.
         /// </summary>
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Failure to set this property should be ignored.")]
         internal static void SetNETFrameworkCompatiblityMode(AppDomainSetup setup, IRunContext runContext)
         {
             try
             {
                 RunConfiguration runConfiguration = XmlRunSettingsUtilities.GetRunConfigurationNode(runContext.RunSettings.SettingsXml);
-                if (null != runConfiguration && Enum.Equals(runConfiguration.TargetFrameworkVersion, FrameworkVersion.Framework40))
+                if (null != runConfiguration && (Enum.Equals(runConfiguration.TargetFrameworkVersion, FrameworkVersion.Framework40) ||
```

Comment:
I don't see any usages of this. Does it belong to the adapter code?

Reply:
Yes, ObjectModel was friend assembly for MSTestV1 adapter.